test/normalized_has_many_test.rb

```diff
@@ -244,4 +244,11 @@ def test_respects_should_use_cache_on_association
       end
     end
   end
+
+  def test_fetch_association_after_adding_to_it
```

Comment:
This test seems to fail for `Item.cache_has_many(:associated_records, embed: true)`, is that expected?

Reply:
Good catch!

I was thinking it was only an issue with id embedding, but I think that is just because the symptoms of the problem were different for id embedding.

I've pushed a commit fixing this for `cache_has_many` `embed: true`